src/java/htsjdk/samtools/util/BlockCompressedInputStream.java

```diff
@@ -355,60 +370,108 @@ public static boolean isValidFile(final InputStream stream)
     }
 
     private static boolean isValidBlockHeader(final byte[] buffer) {
-        return (buffer[0] == BlockCompressedStreamConstants.GZIP_ID1 &&
-                (buffer[1] & 0xFF) == BlockCompressedStreamConstants.GZIP_ID2 &&
-                (buffer[3] & BlockCompressedStreamConstants.GZIP_FLG) != 0 &&
-                buffer[10] == BlockCompressedStreamConstants.GZIP_XLEN &&
-                buffer[12] == BlockCompressedStreamConstants.BGZF_ID1 &&
-                buffer[13] == BlockCompressedStreamConstants.BGZF_ID2);
+        return (buffer[0] == BlockCompressedStreamConstants.GZIP_ID1
+                && (buffer[1] & 0xFF) == BlockCompressedStreamConstants.GZIP_ID2
+                && (buffer[3] & BlockCompressedStreamConstants.GZIP_FLG) != 0
+                && buffer[10] == BlockCompressedStreamConstants.GZIP_XLEN
+                && buffer[12] == BlockCompressedStreamConstants.BGZF_ID1
+                && buffer[13] == BlockCompressedStreamConstants.BGZF_ID2);
     }
 
-    private void readBlock()
-        throws IOException {
-
+    private void readBlock() throws IOException {
+    	mCurrentBlock = nextBlock(getBufferForReuse(mCurrentBlock));
+        mCurrentOffset = 0;
+        checkAndRethrowDecompressionException();
+    }
+    /**
+     * Reads and decompresses the next block
+     * @param bufferAvailableForReuse decompression buffer available for reuse
+     * @return next block in the decompressed stream
+     */
+    protected DecompressedBlock nextBlock(byte[] bufferAvailableForReuse) {
```

Comment:
what's the difference between `nextBlock(byte[] bufferAvailableForReuse)` and `processNextBlock(byte[] bufferAvailableForReuse)`?